addons/mail/static/src/models/message/message.js

```diff
@@ -111,8 +111,10 @@ function factory(dependencies) {
                     data2.parentMessage = insertAndReplace(this.convertData(data.parentMessage));
                 }
             }
-            if ('partner_ids' in data && this.messaging.currentPartner) {
```

Comment:
For API compatibility in stable the handling of the `partner_ids` key should be kept.

Instead of writing `isCurrentPartnerMentioned`, it should update `recipients` too just like the new key does.
As both "replace" will conflict with each other, we need to decide which one to put first.

Probably the old one last so that it takes priority if anybody did anything special with it as custo.